app/controllers/admin/health/patient_referrals_controller.rb

```diff
@@ -168,6 +176,25 @@ def assign_agency
     end
     # rubocop:enable Style/IfInsideElse
 
+    private def assign_agency_inner(patient_referral, agency_id: nil, care_coordinator_id: nil, nurse_care_manager_id: nil)
+      care_staff_id = care_coordinator_id.presence || nurse_care_manager_id.presence
+
+      # agency_id is only present if this is a re-assignment
+      agency_id ||= (Health::AgencyUser.where(user_id: care_staff_id.to_i).pluck(:agency_id).first if care_staff_id.present?)
+
+      return false unless patient_referral.update({ agency_id: agency_id })
```

Comment:
These feels overly complicated, and I don't like return unless an action is successful, essentially this is trying to make a change and then checking success or failure of the change, but the change appears an afterthought.  What is the situation where that update would fail?  A blank agency_id?
Here's my stab at it:
```suggestion
       agency_id = Health::AgencyUser.find_by(user_id: care_staff_id.to_i).&agency_id if agency_id.blank? && care_staff_id.present?
      return false unless agency_id.present?
      
      patient_referral.update(agency_id: agency_id)
```

Reply:
A blank agency ID is fine, that means that the referral is being unassigned from an agency.

`update` fails if there are validation issues. I'm not sure if it would fail due to database transaction failure or anything like that. Based on this blog post, I thought it was best practice to handle failed updates - https://marcgg.com/blog/2020/04/08/rails-update-bang/
but I do not anticipate any validation failures here, so I can remove it if you think that's best!

Reply:
In this case, I guess I'd prefer `update!`, but since you are just saving the agency_id, the only reasons it would fail would be that it was already invalid, or the agency_id is nil (which is probably allowed) so, let's go with silent failure.

Reply:
Okay I removed it in https://github.com/greenriver/hmis-warehouse/pull/2165/commits/d612dc509ab97161dcd79c239bb8a3bbf6a8d30e . It also removes the notices we used to show when update failed, like "Patient could not be assigned."

Reply:
Hmm, removing the error handling wasn't really my intention.  Could we set it to a variable and then check that for the return?  I guess I see `return false unless patient.update(a: :b)` similar to `if(value = patient.do_something) {...}` which I'm not a fan of because the change is hidden within the if.

Reply:
Okay, let me know if the recent update is okay. I see what you're getting at with the control flow, I think it's clearer now.